public/main.js

```diff
@@ -5,6 +5,39 @@ var todoTitle = document.getElementById("new-todo");
 var error = document.getElementById("error");
 var countLabel = document.getElementById("count-label");
 var clearCompleteButton = document.getElementById("clear-complete-button");
+var filterSelector = document.getElementById("todo-filters");
+
+// Filters for items within the todo list; may not be combined
+var listFilters = {
+    // Shows all items
+    "all": function(item) {
+        return true;
+    },
+    // Shows only incomplete items
+    "active": function(item) {
+        return !item.isComplete;
+    },
+    // Shows only complete items
+    "completed": function(item) {
+        return item.isComplete;
+    }
+};
+// The list filter currently being used, shows "all" by default
+var currentFilter = "all";
+
+// Create the list of filter buttons and add them to the page
+var filterButtons = Object.keys(listFilters).map(function(key) {
```

Comment:
This is a bit confusing because of the combination of the comment, the filterButtons variable being unused and the use of the map function. 

If you remove the comment and the filterButtons variable, you can use the following instead to be clearer. 
```
Object.keys(listFilters).forEach(function(key) {
    //Add each filter button to page
   ....
}
```
Note, with foreach, you don't return the filterButton at the end. 
Also, you don't definitely need to do this, but you might consider creating a function for the filter button creation. That way you don't need the comment at all.